app/src/main/java/xyz/gianlu/librespot/android/MainActivity.java

```diff
@@ -1,32 +1,57 @@
 package xyz.gianlu.librespot.android;
 
 import android.os.Bundle;
+import android.util.Log;
 
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 
+import com.spotify.connectstate.Connect;
+
 import java.io.IOException;
 import java.security.GeneralSecurityException;
 
 import xyz.gianlu.librespot.core.Session;
 import xyz.gianlu.librespot.mercury.MercuryClient;
+import xyz.gianlu.librespot.player.Player;
+import xyz.gianlu.librespot.player.PlayerConfiguration;
 
 public class MainActivity extends AppCompatActivity {
+    private static final String TAG = "Main";
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         new Thread(() -> {
+            Session session;
             try {
-                Session session = new Session.Builder()
-                        .userPass("test123", "test123")
-                        .create();
-
-                System.out.println("Logged in as: " + session.apWelcome().getCanonicalUsername());
-            } catch (IOException | GeneralSecurityException | Session.SpotifyAuthenticationException | MercuryClient.MercuryException ex) {
-                ex.printStackTrace();
+                Session.Configuration conf = new Session.Configuration.Builder()
+                        .setStoreCredentials(false)
+                        .setCacheEnabled(false).build();
+                session = new Session.Builder(conf)
+                        .setPreferredLocale("en")
+                        .setDeviceType(Connect.DeviceType.SMARTPHONE)
+                        .setDeviceName("librespot-java")
+                        .userPass("user", "password")
+                        .setDeviceId(null).create();
+
+                Log.i(TAG, "Logged in as: " + session.apWelcome().getCanonicalUsername());
+            } catch (IOException |
+                    GeneralSecurityException |
+                    Session.SpotifyAuthenticationException |
+                    MercuryClient.MercuryException ex) {
+                Log.e(TAG, "Session creation failed: ", ex);
+                return;
             }
+            PlayerConfiguration configuration = new PlayerConfiguration.Builder()
+                    .setOutput(PlayerConfiguration.AudioOutput.CUSTOM)
+                    .setOutputClass("xyz.gianlu.librespot.android.sink.AndroidSinkOutput")
+                    .setOutputClassParams(new String[0])
```

Comment:
Current Builder Implementation expects an empty array if no parameters provided. Seems like a bug to me.

Reply:
Definitely, might have slipped in with https://github.com/librespot-org/librespot-java/pull/346. Fixed in https://github.com/librespot-org/librespot-java/commit/332d334b8c0332827ef6d122a283dce5e7660535